### DISCUSSION TITLE
Minimal Sound

### DISCUSSION THREAD

**Discussion by TwoPigsToTheWind** [slu4coder/Minimal-64x4-Home-Computer#9] [Ideas]
![image](https://github.com/slu4coder/Minimal-64x4-Home-Computer/assets/169104967/776ff8ec-1a71-482e-ac3e-e0fc2cfb52c1)
![Screenshot_20240508_114052](https://github.com/slu4coder/Minimal-64x4-Home-Computer/assets/169104967/47d019a1-41b3-4d7a-864b-b6be09b81024)

Possible use of the expansion port based on George Foot's tone generator design https://www.youtube.com/watch?v=LNjJJTeZ1Wg

Two 74HC688s and a 6 pos DIP switch are used to set the address range that the registers on the card are mapped to. When the address lines are with in that range the value on the data bus is stored to one of four 74HC734s that feed into four divide by N 74HC40103 counters, these in turn drive a pair of D flip flop 74HC74s the output of which is passed thru a resistor to reduce the voltage before outputting to a headphone jack. Should be enough to drive a pair of headphones. May add an on board pizo buzzer as an alternative to external speakers. 

Plan to test it out on a breadboard before ordering the board of course. I don't understand the timing for the address and bus lines well enough to say 'yes this will work' once it's built out on breadboards if there are timing issues we'll go from there.

**Comment by bobbarnes1981** [comment]
I would be interested to see a video of this when you get it working

**Reply by TwoPigsToTheWind** [reply]
can do

**Comment by nateo87** [comment]
Wow, love that you're pursing this. I just started soldering my Minimal together; excited to see that folks are already working on expansions for it!

**Comment by hans61** [comment]
Very interesting circuit, I didn't know that before.
I tested another way for the sound. I have a SN76489. It's the same sound as the BBC Micro. It's still available cheaply.
I'm just at the beginning. I would like to add a Kempston joystick port and an SD card.
Here is my current status:
https://github.com/hans61/Minimal-64x4/tree/main/hardware/SN76489
very simple video:
https://www.youtube.com/watch?v=yanbZCrXf6M

**Reply by nateo87** [reply]
Just followed it - looks really cool. Is there a standard way to have expansion cards organized at this point, backplane, etc?

**Reply by TwoPigsToTheWind** [reply]
the current setup is one physical port that is stackable. That one port could be used to connect to a passive expansion backplane. Could also change and expand the main boar to include more physical ports.

**Comment by TwoPigsToTheWind** [comment]
I had though about using a PSG like the SN76489 but thought a 74HC based solution was more in line with the original theme of the Minimal. Both great solutions though

**Comment by slu4coder** [comment]
Guys, it's just amazing to see what you are doing here. Both the awesome sound player by Hans as well as some more "Minimal'istic" beep solutions :-)

**Comment by hans61** [comment]
I've already done something in the direction of the backplane. It was actually just meant for me.
I just put it on my git for anyone interested.
https://github.com/hans61/Minimal-64x4/tree/main/hardware/BusExpansion

I'm also trying out an SPI with TTL for connecting an SD card.
I haven't had any success with the 74HC299 so far. I'm now trying with 74HC165 and 74HC595.
I didn't want Bit Bang.

A few ideas come to mind. RAM banking or whether you can get a vc1541 connected.
I still have some from the C64.

I don't want an AVR or anything like that on the minimal. But I wouldn't want to exclude all IC from the 1980s.
Nevertheless, I have so far avoided using a PAL/GAL or small CLPD.

**Comment by hans61** [comment]
I tested a revised version of my sound extension

https://github.com/hans61/Minimal-64x4/tree/main/hardware/SN76489

**Reply by TwoPigsToTheWind** [reply]
Looks great!
